tests/unit/test_client.py

```diff
@@ -7775,6 +7775,43 @@ def test_load_table_from_json_w_invalid_job_config(self):
         err_msg = str(exc.value)
         assert "Expected an instance of LoadJobConfig" in err_msg
 
+    def test_load_table_from_json_unicode_emoji_data_case(self):
+        from google.cloud.bigquery.client import _DEFAULT_NUM_RETRIES
+        from google.cloud.bigquery import job
+
+        client = self._make_client()
+
+        json_row = {"emoji": "\U0001F3E6"}
+        json_rows = [
+            json_row
+        ]
+
+        load_patch = mock.patch(
+            "google.cloud.bigquery.client.Client.load_table_from_file", autospec=True
+        )
+
+        with load_patch as load_table_from_file:
+            client.load_table_from_json(json_rows, self.TABLE_REF)
+
+        load_table_from_file.assert_called_once_with(
+            client,
+            mock.ANY,
+            self.TABLE_REF,
+            size=mock.ANY,
+            num_retries=_DEFAULT_NUM_RETRIES,
+            job_id=mock.ANY,
+            job_id_prefix=None,
+            location=client.location,
+            project=client.project,
+            job_config=mock.ANY,
+            timeout=None,
+        )
+
+        sent_data_file = load_table_from_file.mock_calls[0][1][1]
+
+        # make sure json_row's unicode characters are only encoded one time
+        assert sent_data_file.getvalue() == json.dumps(json_row, ensure_ascii=False).encode()
```

Comment:
It would be cleaner to assert that the emoji is present, rather than using the same logic as the test.  E.g.:
```suggestion
        sent_data_file.seek(0)  # rewind
        sent_row = json.load(sent_data_file)
        assert sent_row["emoji"] == emoji
```

Reply:
Hmmm, my proposed change here is defective, because `json.load()` returns the correct value whether or not the `sent_data_file` was generated with `ensure_ascii=False`.

Reply:
So, for a better choice, let's format the bytes we actually expect to see on the wire.
```python
        expected_bytes = b'{"emoji": "' + emoji.encode("utf8") + b'"}'
        assert sent_data_file.getvalue() == expected_bytes
```

Reply:
Thank you. It makes sense. Originally I thought of it this way but somehow I did not. I have pushed a new commit for this.